addons/stock_account/models/account_move.py

```diff
@@ -112,7 +112,7 @@ def _stock_account_prepare_anglo_saxon_out_lines_vals(self):
             for line in move.invoice_line_ids:
 
                 # Filter out lines being not eligible for COGS.
-                if line.product_id.type != 'product' or line.product_id.valuation != 'real_time':
+                if line.product_id.type != 'product' or line.product_id.with_company(line.company_id).valuation != 'real_time':
```

Comment:
This is the line that would prevent cogs to be generated if company B wasn't using real_time valuation for its product.